[DOCS-5848] Add Java instructions for CI Visibility ITR (#18777)
DataDog/documentation: content/en/continuous_integration/intelligent_test_runner/java.md

---
title: Intelligent Test Runner for Java

> kind: documentation

Reviewer: ```suggestion
kind: documentation
is_beta: true
```

further_reading:
    - link: "/continuous_integration/tests"
      tag: "Documentation"
      text: "Explore Test Results and Performance"
    - link: "/continuous_integration/troubleshooting/"
      tag: "Documentation"
      text: "Troubleshooting CI"
---

> <div class="alert alert-warning">Intelligent Test Runner for Java is a beta product.</div>

Reviewer: ```suggestion
{{< callout url="#" >}}Intelligent Test Runner for Java is a beta product.{{< /callout >}} 
```

## Compatibility

> Intelligent Test Runner is supported in `dd-java-agent >= 1.18.0`

Reviewer: ```suggestion
Intelligent Test Runner is supported in `dd-java-agent >= 1.18.0`.
```

## Setup

Prior to setting up Intelligent Test Runner, set up [Test Visibility for Java][1]. If you are reporting data through the Agent, use v6.40+/v7.40+.

To enable Intelligent Test Runner, set the following environment variables:

`DD_APPLICATION_KEY` (Required)
: The [Datadog Application key][2] used to query the tests to be skipped.<br/>

> If you are reporting data through the Agent, then it is the agent process that will require the key.<br/>
> If you are using agentless mode, then they key will be used by the tracer process.<br/>

Reviewer: ```suggestion
If you are reporting data through the Agent, then the agent process requires the key.<br/>
If you are using Agentless mode, then the tracer process uses the key.<br/>
```

**Default**: `(empty)`<br/>

`DD_CIVISIBILITY_JACOCO_PLUGIN_VERSION` (Optional)

> : Intelligent test runner requires [Jacoco][4] for collecting code coverage.<br/>

Reviewer: ```suggestion
: Intelligent Test Runner requires [Jacoco][4] for collecting code coverage.<br/>
```

> Set this variable to a valid Jacoco version (e.g. `0.8.10`) if you want the tracer run your build with Jacoco injected.<br/>

Reviewer: ```suggestion
Set this variable to a valid Jacoco version (such as `0.8.10`) if you want the tracer to run your build with Jacoco injected.<br/>
```

Omit it if your project already has Jacoco configured.<br/>
**Default**: `(empty)`

`DD_CIVISIBILITY_JACOCO_PLUGIN_INCLUDES` (Optional)
: Use this variable to configure the list of packages for which code coverage should be collected.<br/>

> It is optional, but setting this helps to reduce overhead by avoiding coverage collection for core JDK classes and 3rd party dependencies.<br/>

Reviewer: ```suggestion
While this variable is optional, setting this helps reduce overhead by avoiding coverage collection for core JDK classes and third-party dependencies.<br/>
```
